src/features/web-vitals.js

```diff
@@ -0,0 +1,11 @@
+import ContentFeature from '../content-feature'
+
+export default class WebVitals extends ContentFeature {
+    init () {
+        this.messaging.subscribe('getVitals', () => {
+            const paintResources = performance.getEntriesByType('paint')
+            const firstPaint = paintResources.find((entry) => entry.name === 'first-contentful-paint')
+            this.messaging.notify('vitalsResult', { vitals: [firstPaint?.startTime] })
```

Comment:
if the entry is absent then this will send `vitals: [null]` (array with a single `null` item in it) as the message params - not sure if that's how you intended to handle this situation on the native side?

if not, you could just wrap in an `if` condition and only send if `firstPaint` is found

Reply:
Its not a problem on the native side. It just parses to `nil` eventually. But I can change it if it feels wrong on the JS side.

Reply:
yeah, I know this is super picky - but very soon we're landing something that enforces all boundary messages like this to be defined in a schema, so we wouldn't want to enable what is effectively `(string|null)[]`.

anyway, if you just do this instead it'll send an empty array 👌🏻 

```js
const paintResources = performance.getEntriesByType('paint')
const vitals = paintResources.filter((entry) => entry.name === 'first-contentful-paint')
this.messaging.notify('vitalsResult', { vitals })
```

Reply:
actually scratch that

or, to prevent needing to pluck out 'startTime' in that example, this instead?

```js
const paintResources = performance.getEntriesByType('paint')
const firstPaint = paintResources.filter((entry) => entry.name === 'first-contentful-paint')
const vitals = firstPaint ? [firstPaint.startTime] : []
this.messaging.notify('vitalsResult', { vitals })
```